test/functional/feature_framework_startup_failures.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Verify framework startup failures only raise one exception since
+multiple exceptions being raised muddies the waters of what actually
+went wrong. We should maintain this bar of only raising one exception as
+long as additional maintenance and complexity is low.
+
+Test relaunches itself into child processes in order to trigger failures
+without the parent process' BitcoinTestFramework also failing.
+"""
+
+from test_framework.util import rpc_port
+from test_framework.test_framework import BitcoinTestFramework
+
+import re
+import subprocess
+import sys
+import time
+
+class FeatureFrameworkStartupFailures(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_network(self):
+        # Don't start the node yet, as we want to measure during run_test.
+        self.add_nodes(self.num_nodes, self.extra_args)
+
+    # Launches a child test process running this same file, but instantiating
+    # a child test, and verifies that we (only) raise the expected exception.
+    def _verify_startup_failure(self, test, internal_args, expected_exception):
+        args = [
+            sys.executable,
+            __file__,
+            f"--cachedir={self.options.cachedir}",
+            f"--portseed={self.options.port_seed}",
+            f"--internal_test={test.__name__}",
+        ] + internal_args
+        if self.options.nocleanup:
+            args.append("--nocleanup")
+        if self.options.tmpdir:
+            args.append(f"--tmpdir={self.options.tmpdir}/{hash(expected_exception)}"),
+        if self.options.randomseed is not None:
+            args.append(f"--randomseed={self.options.randomseed + hash(expected_exception)}")
+        if self.options.timeout_factor is not None:
+            args.append(f"--timeout-factor={self.options.timeout_factor}")
+
+        try:
+            output = subprocess.run(args,
+                                    encoding="utf-8",
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.STDOUT,
+                                    timeout=60 * self.options.timeout_factor).stdout
+        except subprocess.TimeoutExpired as e:
+            # e.output is returned as bytes on Linux and str on Windows.
+            child_output = e.output if isinstance(e.output, str) else e.output.decode("utf-8")
```

Comment:
https://docs.python.org/3/library/subprocess.html#subprocess.TimeoutExpired.output claims:
> This is always [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) when any output was captured

Can we simplify to:

```suggestion
            child_output = e.output.decode("utf-8") if e.output else ""
```

?

Reply:
Pushed an alternative, still using `print`. When I tried `sys.stderr.buffer.write` that data ended up at the end of the failure output, instead of when the exception was caught.